dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataFlusher.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.upload
+
+import com.datadog.android.core.internal.net.DataUploader
+import com.datadog.android.core.internal.persistence.PayloadDecoration
+import com.datadog.android.core.internal.persistence.file.FileHandler
+import com.datadog.android.core.internal.persistence.file.advanced.ConsentAwareFileOrchestrator
+
+internal class DataFlusher(
+    internal val fileOrchestrator: ConsentAwareFileOrchestrator,
```

Comment:
Why couple the flushin mecanism with the consent tracking one? 
Instead it'd be better to have a `FileOrchestrator.getFlushableFiles()` instead, and let the `ConsentAwareFileOrchestrator` decide that it means granted only.